vms/platformvm/state/state.go

```diff
@@ -1944,3 +2010,50 @@ func (s *state) writeMetadata() error {
 	}
 	return nil
 }
+
+// TODO: Remove after next hard-fork when everybody has run this.
+func (s *state) populateBlockHeightIndex() error {
+	blk, _, err := s.GetStatelessBlock(s.lastAccepted)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve last accepted block from local disk: %w", err)
+	}
+
+	heightKey := database.PackUInt64(blk.Height())
+	if _, err := database.GetID(s.blockIDDB, heightKey); err == nil {
+		// If the last accepted block is in [s.blockIDDB], no backfilling is required.
+		return nil
+	}
+
+	blockIterator := s.blockDB.NewIterator()
+	defer blockIterator.Release()
+	for blockIterator.Next() {
```

Comment:
question: do we need this process to be synchronous, i.e. do we need the full vm to be declared initialized after it completes? If not, should we consider running it asynchronously and let the VM complete initialization while reindexing is ongoing?
I ask because we had a similar issue in the past with proposerVM ( a faulty height index required reindexing) and we run it async and cap the max rate with some sleeps to avoid choking the node,

Reply:
It doesn't _need_ to be synchronous but I wanted to avoid asynchronous since the resulting code will be less obviously correct and more error-prone. A 40s delay for initialization seems short enough to go forward with the synchronous approach

Reply:
> 40s delay for initialization seems

Should we add logging, tracing, or metrics to track this? Can be done in a separate PR but at least logging to start with will be great.

Reply:
added a log